Source/RequestBuilder.swift

```diff
@@ -475,7 +475,14 @@ class RequestBuilder: NSObject {
     }
     
     func getExternalUserIds() -> [[AnyHashable: Any]]? {
-        let externalUserIdArray : [ExternalUserId] = Prebid.shared.externalUserIdArray
+       
+        var externalUserIdArray = [ExternalUserId]()
+        if Prebid.shared.externalUserIdArray.count != 0 {
+            externalUserIdArray = Prebid.shared.externalUserIdArray
+        }
+        else if Targeting.shared.externalUserIds.count != 0{
```

Comment:
why is the externalUserId in 2 places? Prebid.shared & Targeting.shared?

Reply:
ExternalUserId can be set either by Prebid.shared.externalUserIdArray API or by local storage option from Targeting.shared getter & setter.
And based on details here https://github.com/prebid/prebid-mobile-ios/issues/347 we'll give first priority to API and after that to local storage